tableauserverclient/server/endpoint/permissions_endpoint.py

```diff
@@ -0,0 +1,73 @@
+import logging
+
+from .. import RequestFactory, PermissionsItem
+
+from .endpoint import Endpoint, api
+from .exceptions import MissingRequiredFieldError
+
+
+logger = logging.getLogger(__name__)
+
+
+class _PermissionsEndpoint(Endpoint):
+    ''' Adds permission model to another endpoint
+
+    Tableau permissions model is identical between objects but they are nested under
+    the parent object endpoint (i.e. permissions for workbooks are under
+    /workbooks/:id/permission).  This class is meant to be instantated inside a
+    parent endpoint which has these supported endpoints
+    '''
+    def __init__(self, parent_srv, owner_baseurl):
+        super(_PermissionsEndpoint, self).__init__(parent_srv)
+
+        # owner_baseurl is the baseurl of the parent.  The MUST be a lambda
+        # since we don't know the full site URL until we sign in.  If
+        # populated without, we will get a sign-in error
+        self.owner_baseurl = owner_baseurl
+
+    def update(self, item, permission_item):
+        url = '{0}/{1}/permissions'.format(self.owner_baseurl(), item.id)
```

Comment:
project permissions definitions also includes default permissions for workbook and datasources access, it would be nice to have an url separation e.g. `url = '{0}/{1}/default-permissions/{2}'.format(self.owner_baseurl(),                                                 item.id, default)` (where default is 'workbooks' or 'datasources')